app/services/hyrax/autopopulation/create_work.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Hyrax
+  module Autopopulation
+    class CreateWork
+      attr_accessor :attributes, :user, :cname, :pdf_url
+
+      # params [attributes] is a hash constructed from a crossref response that has been crosswalked by a custom Bolognese writer
+      # example using the doi 10.1117/12.2004063
+      # response = Bolognese::Metadata.new(input: '10.1117/12.2004063', from: "crossref")
+      # attributes = response.build_work_actor_attributes
+      #
+      def initialize(attributes, user, cname = nil)
+        AccountElevator.switch!(cname) if Rails.application.config.hyrax_autopopulation.storage_type == "activerecord"
+
+        # This is not a property of any work, so it cannot be passed to the actor
+        @pdf_url = attributes.delete(:unpaywall_pdf_url)
+
+        pdf_url.present? && uploaded_file.present? && attributes.merge!(uploaded_files: uploaded_file)
+
+        @attributes = attributes
+        @user = user
+        @cname = cname
+      end
+
+      def save
+        # saves work
+        # This returns true when the work is aved or false
+        actor.create(actor_environment)
+      end
+
+      private
+
+        def actor
+          Hyrax::CurationConcern.actor
+        end
+
+        def actor_environment
+          if Rails.application.config.hyrax_autopopulation.storage_type == "redis"
+            # Remove fields not defined by Hyrax
+            keys = %i[date_published editor]
+            new_attributes = attributes.except(*keys)
+            @_actor_environemnt ||= Hyrax::Actors::Environment.new(GenericWork.new, ::Ability.new(user), new_attributes)
+          else
+            @_actor_environemnt ||= Hyrax::Actors::Environment.new(GenericWork.new, ::Ability.new(user), attributes)
+          end
+        end
```

Comment:
This method is a little messy. 

Using the `begin` method means you can avoid writing `@_actor_environemnt ||=` twice. 
```ruby
        def actor_environment
          @_actor_environemnt ||= begin
              if Rails.application.config.hyrax_autopopulation.storage_type == "redis"        
                 # Remove fields not defined by Hyrax
                 attrs = attributes.except(*%i[date_published editor])
              else
                 attrs = attributes
              end

               Hyrax::Actors::Environment.new(GenericWork.new, ::Ability.new(user), attrs)
           end
        end
```

Reply:
Thanks, I think it was a case of  avoiding **@_actor_evironment** twice or avoiding **attrs** twice.